### PR TITLE
Fix unit tests for .net core 3.1

### PR DESCRIPTION
**Pull request by philipktlin**
Fix most of unit tests (except ConditionCompilerTest.TestSuccess and WorkloadClientTest.GetServerCertificatesFromEdgeletShouldReturnCert).  Since this PR is getting too big, we can continue to fix in another PR.

### CODE REVIEW

File: edge-hub/test/Microsoft.Azure.Devices.Routing.Core.Test/query/ConditionCompilerTest.cs [hunk 1]
```diff
@@ -602,7 +602,8 @@ static class TestSuccessDataSource
                 new object[] { "as_number(\"3\") = 3", Bool.True },
                 new object[] { "AS_NUMBER(\"3\") = 3", Bool.True },
                 new object[] { "AS_NUMBER('10e10') = 10e10", Bool.True },
-                new object[] { "AS_NUMBER('10e1000') = 1/0", Bool.Undefined },
+                new object[] { "AS_NUMBER('10e1000') = 10e1000", Bool.True },
+                new object[] { "AS_NUMBER('10e1000') = 1/0", Bool.True },
```
Comment (philipktlin) [inline edge-hub/test/Microsoft.Azure.Devices.Routing.Core.Test/query/ConditionCompilerTest.cs:606]: 10e1000 throws exception in .Net Core 2.2, but it doesn't in .Net Core 3.1.  For details, read https://docs.microsoft.com/en-us/dotnet/core/compatibility/2.2-3.1#floating-point-parsing-operations-no-longer-fail-or-throw-an-overflowexception

File: edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs [hunk 2]
```diff
@@ -555,7 +555,7 @@ public async System.Threading.Tasks.Task<CertificateResponse> CreateServerCertif
                         ProcessResponse(client_, response_);
     
                         var status_ = ((int)response_.StatusCode).ToString();
-                        if (status_ == "201") 
+                        if (status_ == "200")
```
Comment (philipktlin) [inline edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs]: @and-rewsmith WorkloadClientTest.GetServerCertificatesFromEdgeletShouldReturnCert failed if 201 is used here.  I found that master use 200,  do you know why generated code changed to 201?

**Comment by philipktlin** [comment]
@and-rewsmith  do you know why ModuleManagementHttpClientTest Versions class variable defines only 3 versions, not 4 (2019-11-05 missing)? does it means we doesn't need to verify 2019-11-05 version for any tests in ModuleManagementHttpClientTest?

File: Microsoft.Azure.Devices.Edge.sln [hunk 3]
```diff
@@ -219,15 +219,15 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MetricsValidator", "test\mo
...
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "samples", "samples", "{C077918C-9AD4-4F01-ADED-4B8B6AB784A2}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "dotnet", "dotnet", "{A6D8677F-DB76-459E-B6DE-110AFCEF7F08}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "EdgeDownstreamDevice", "samples\dotnet\EdgeDownstreamDevice\EdgeDownstreamDevice.csproj", "{7546DB89-643A-477F-AEE3-FA05BDCE4ACD}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "EdgeDownstreamDevice", "samples\dotnet\EdgeDownstreamDevice\EdgeDownstreamDevice.csproj", "{7546DB89-643A-477F-AEE3-FA05BDCE4ACD}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "EdgeX509AuthDownstreamDevice", "samples\dotnet\EdgeX509AuthDownstreamDevice\EdgeX509AuthDownstreamDevice.csproj", "{9E809F78-0EFF-4860-94D1-8D0EF1A9C1F1}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "EdgeX509AuthDownstreamDevice", "samples\dotnet\EdgeX509AuthDownstreamDevice\EdgeX509AuthDownstreamDevice.csproj", "{9E809F78-0EFF-4860-94D1-8D0EF1A9C1F1}"
```
Comment (and-rewsmith) [inline Microsoft.Azure.Devices.Edge.sln:230]: What is this change? Did you remove and re-add to solution?

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Http/HttpProtocolHead.cs [hunk 4]
```diff
@@ -3,37 +3,37 @@ namespace Microsoft.Azure.Devices.Edge.Hub.Http
 {
     using System.Threading;
     using System.Threading.Tasks;
-    using Microsoft.AspNetCore.Hosting;
     using Microsoft.Azure.Devices.Edge.Hub.Core;
     using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Hosting;
```
Comment (and-rewsmith) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Http/HttpProtocolHead.cs:8]: Why this change?

File: edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs (hunk 2 above)
Comment (and-rewsmith) [reply edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs]: No, the only difference is that I generated with the dotnet 3 option. We should find a way to document these changes so that the next time someone generates the code they don't face the same headache.

Comment (and-rewsmith) [reply edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs]: We should also change on line 464, right?

File: Microsoft.Azure.Devices.Edge.sln (hunk 3 above)
Comment (philipktlin) [reply Microsoft.Azure.Devices.Edge.sln:230]: This is managed by Visual studio 2019.  And it looks good to me as these projects have the same parent node.  Would you open it in VS Code and build it without any problem?

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Http/HttpProtocolHead.cs (hunk 4 above)
Comment (philipktlin) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Http/HttpProtocolHead.cs:8]: Check out Hosting.cs changes, as it is recommended to replace WebHostBuilder by HostBuilder (https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.1&tabs=visual-studio#hostbuilder-replaces-webhostbuilder).

File: test/modules/ModuleRestarter/ModuleRestarter.csproj [hunk 5]
```diff
@@ -16,12 +16,12 @@
...
-    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="2.2.4" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="2.2.4" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="2.2.0" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="3.1.3" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="3.1.3" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="3.1.3" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="3.1.3" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="3.1.3" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.1.3 />
```
Comment (and-rewsmith) [inline test/modules/ModuleRestarter/ModuleRestarter.csproj]: missing " I can fix

File: edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs (hunk 2 above)
Comment (philipktlin) [reply edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs]: I am thinking we should update server side to return 201 instead of 200.  Would you please have to work on this.  I think I should revert this change, and handle this in another PR.  What do you think?

File: test/modules/ModuleRestarter/ModuleRestarter.csproj (hunk 5 above)
Comment (philipktlin) [reply test/modules/ModuleRestarter/ModuleRestarter.csproj:24]: interesting.  I can see it in my local copy.

File: edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs (hunk 2 above)
Comment (and-rewsmith) [reply edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs]: I agree with separate PR and can help make this change on the server side (edgelet). I think we should first prioritize the CI, e2e, and release build though.

File: Microsoft.Azure.Devices.Edge.sln (hunk 3 above)
Comment (and-rewsmith) [reply Microsoft.Azure.Devices.Edge.sln:230]: Yep, builds fine.

File: edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs (hunk 2 above)
Comment (philipktlin) [reply edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs]: Reverted.
